### DISCUSSION TITLE
Swagger ignores JsonInclude.Include.NON_EMPTY for not-nullable fields

### DISCUSSION THREAD

**Discussion by DerJochen** [quarkusio/quarkus#33195] [Q&A] [answered]
In my Kotlin code I have a data class with a not-nullable List field (see below). Eventhough the field is not included in the JSON if the List is empty, Swagger marks the field as required (red asterisk).
If I make the field nullable (adding a question mark) the API documentation is correct, but I would like to not change my implementation just to get the correct documentation. Is there another option to fix this?

```
@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
data class MyClass(
	@field:Schema(required = false, minItems = 1)
	val data: List<Data> = emptyList(),
)

```

[a comment by an automated account is not shown]

**Comment by MikeEdgar** [comment, marked as answer]
@DerJochen , the scanner is making it required based on the presence of `@NotNull` on the field and the getter generated by Kotlin. When setting the `required` attribute, it is taking both the field and the generated method into account, so adding an annotation to apply to the getter should fix this in your case.

```kotlin
@get:Schema(required = false)
```

**Reply by DerJochen** [reply]
Thanks! That works.
